src/AD.BaseTypes.Core/Extensions/TypeExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace AD.BaseTypes.Extensions;
+
+/// <summary>
+/// Type extensions
+/// </summary>
+public static class TypeExtensions
+{
+    /// <summary>
+    /// Determines whether an interface is implemented by the current <see cref="Type"/>.
+    /// </summary>
+    /// <param name="type">The type to be verified.</param>
+    /// <param name="interface">The interface to be searched for.</param>
+    /// <returns><see langword="true" /> when the type implements the interface.</returns>
+    public static bool Implements(this Type type, Type @interface)
```

Comment:
Can we make "Implements" BaseType-specific? Like "ImplementsIBaseType"?

Reply:
Yeah sure no problems

Reply:
@Andreas-Dorfer I renamed the function and adapted the tests.